src/auth/guard/auth.guard.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { OAuth2Client } from 'google-auth-library';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  private googleClient = new OAuth2Client(process.env.GOOGLE_CLIENT_ID);
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
```

Comment:
- Technically we dont need to guard auth routes
- Use this implementation inside the service class to fetch user information using the provided token